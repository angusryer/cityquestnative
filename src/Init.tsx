import React, { useState, useEffect } from "react";
import { registerRootComponent } from "expo";
import { UserStatus } from "./models/statusEnums";
import { Splash } from "./ui/containers/Splash";
import { Menu } from "./ui/containers/Menu";
import { Auth } from "./ui/containers/Auth";
import { initializeGame } from "./logic/init/gameLogic";
import { GameError } from "./logic/error/exceptionLogic";

export default registerRootComponent(Init);

export function Init() {
	const [appIsReady, setAppIsReady] = useState<boolean>(false);
	const [authIsNeeded, setAuthIsNeeded] = useState<boolean>(false);

	const prepareAppResources = async (): Promise<void> => {
		const response = await initializeGame();
		switch (response?.type) {
			case UserStatus.NOT_AUTHENTICATED:
				setAuthIsNeeded(true);
			case UserStatus.AUTHENTICATED:
				setAppIsReady(true);
				break;
			default:
				throw new GameError(response?.message).logError();
		}
	};

	useEffect(() => {
		prepareAppResources();
	}, []);

	if (authIsNeeded) return <Auth />;
	if (appIsReady) return <Menu />;
	else return <Splash />;
}
