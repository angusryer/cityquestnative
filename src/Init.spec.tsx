import React from "react";
import renderer from "react-test-renderer";
import { Init } from "./Init";

describe("Game Initialization", () => {
	it("<Init /> matches previous snapshot", () => {
		const tree: any = renderer.create(<Init />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
