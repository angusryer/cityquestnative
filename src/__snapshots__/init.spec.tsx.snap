// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Initialization <Init /> matches previous snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text>
    This is the Splash Screen
  </Text>
</View>
`;
